corehq/ex-submodules/soil/__init__.py

```diff
@@ -257,3 +258,60 @@ def create(cls, path, **kwargs):
         additional arguments to pass through to the constructor.
         """
         return cls(filename=path, **kwargs)
+
+
+class BlobDownload(DownloadBase):
+    """
+    Download that lives in the blobdb
+    """
+    has_file = True
+
+    def __init__(self, identifier, mimetype="text/plain",
+                 content_disposition='attachment; filename="download.txt"',
+                 transfer_encoding=None, extras=None, download_id=None, cache_backend=SOIL_DEFAULT_CACHE,
+                 content_type=None, bucket=DEFAULT_BUCKET):
+
+        super(BlobDownload, self).__init__(
+            mimetype=content_type if content_type else mimetype,
+            content_disposition=content_disposition,
+            transfer_encoding=transfer_encoding,
+            extras=extras,
+            download_id=download_id,
+            cache_backend=cache_backend
+        )
+        self.identifier = identifier
+        self.bucket = bucket
+        self.blobdb = get_blob_db()
+
+    def get_filename(self):
+        return self.identifier
+
+    def get_content(self):
+        file_obj = None
+        try:
+            file_obj = self.blobdb.get(self.identifier, self.bucket).read()
+            return file_obj.read()
```

Comment:
yeah... i only implemented it because the API defined it. i think i'm actually going to just raise a notimplementederror because we shouldn't be calling `get_content` with the blobdownload